lib/course/data.dart

```diff
@@ -19,7 +20,16 @@ class Course implements Entity {
         assert(name != null),
         assert(description != null),
         assert(teachers != null),
-        assert(color != null);
+        assert(color != null),
+        lessons = LazyIds<Lesson>(
+          collectionId: 'lessons of course $id',
+          fetcher: () async => (await fetchJsonListFrom('lessons?courseId=$id'))
+              .map((data) => Lesson.fromJson(data)),
```

Comment:
What about a static fetcher method on `Lesson` where you can provide `courseId` as an optional parameter? This would hide implementation details